### ISSUE TITLE
Python: Enable auto function/tool calls

### ISSUE THREAD

**Issue by matthewbolanos** [microsoft/semantic-kernel#4553]

**Comment by matthewbolanos** [comment]
@bashimr, it looks like the feature branch we are working in may be ready for you to start working on this feature. Reach out to @eavanvalkenburg.

**Comment by bashimr** [comment]
@matthewbolanos @eavanvalkenburg. Bringing my offline conversation with @moonbox3 here. I'll be happy to take on this task when the team is ready. In the meantime, I'll go through the backlog to see if I can pick up another task. 

From @moonbox3:
Hi Imran, thanks for reaching out. I was looking at that task more after our meeting and it makes more sense to work on it once the kernel arguments PR is completed. We’ll probably need another week, or a bit longer, to get that and some other work in.

**Comment by moonbox3** [comment]
@bashimr, @eavanvalkenburg corrected me and said if you work in our current feature branch (that we talked about in the Python dev meeting yesterday) then it should be fine to work on this in parallel.

**Comment by bashimr** [comment]
Sounds good @moonbox3. Do you have any guidance for me to get started? Or should I just fork the code and start looking at it and ask questions?

**Comment by moonbox3** [comment]
@bashimr let's find some time to sync with @eavanvalkenburg. He is +6 hours from EST so perhaps we can find some time around 9:30 or 10 am EST early next week?

**Comment by bashimr** [comment]
That works for me. I'll connect via email for a calendar invite.
